### PR TITLE
[REF] web: replace jQuery UI Datepicker usage in Calendar view

### PR DESCRIPTION
**Pull request by pparidans**
This commit removes the jQuery UI Datepicker usage for the sidepanel's
mini-calendar in the Calendar View and replace it by our own
DateTimePicker component.

On the visual side, the styling has been aligned to the DateTimePicker
one and takes a bit more space to allow a better readability.

It also adds two props to the DateTimePicker:
- showWeekNumbers : allow to show or hide the week numbers' column, default to true.
- daysOfWeekFormat : allow to choose between the "short" or "narrow" (1
  letter) format for the week's days column header, default to "short".

task-3439226

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Comment by pparidans** [comment]
Waiting for https://github.com/odoo/odoo/pull/134768 to be merged.

File: addons/web/static/src/core/datetime/datetime_picker.js [hunk 1]
```diff
@@ -14,6 +14,8 @@ import {
 import { localization } from "../l10n/localization";
 import { ensureArray } from "../utils/arrays";
 
+const { Info } = luxon;
```
Comment (sdegueldre) [inline addons/web/static/src/core/datetime/datetime_picker.js:17]: Merge with declaration line 79

File: addons/web/static/src/core/datetime/datetime_picker.js [hunk 2]
```diff
@@ -425,7 +436,7 @@ export class DateTimePicker extends Component {
             additionalMonth: this.additionalMonth,
             maxDate: this.maxDate,
             minDate: this.minDate,
-            showWeekNumbers: !this.props.range,
+            showWeekNumbers: this.props.showWeekNumbers && !this.props.range,
```
Comment (sdegueldre) [inline addons/web/static/src/core/datetime/datetime_picker.js:439]: I'd think that the props should take priority
```suggestion
            showWeekNumbers: this.props.showWeekNumbers ?? !this.props.range,
```

Comment (pparidans) [reply addons/web/static/src/core/datetime/datetime_picker.js:439]: makes sense

**Review by adr-odoo — APPROVED** [review]
@robodoo r+

**Pull request closed without merging**